src/main/java/seedu/address/logic/parser/util/GeneralParserUtil.java

```diff
@@ -132,4 +132,25 @@ public static Index parseIndex(String oneBasedIndex) throws ParseException {
         return Index.fromOneBased(Integer.parseInt(trimmedIndex));
     }
 
+    /**
+     * todo javadocs
+     */
+    public static void checkCommandDetailsIsNotBlank(String commandDetails, String itemType, String message)
+        throws ParseException {
+
+        if (commandDetails.isBlank()) {
+            switch (itemType) {
```

Comment:
I was wondering if there is a need for a switch statement here? Correct me if I am mistaken, but this just checks if the user provides an string query for to search for? Is the intent of this function for a more tailored error messaged based on the item type queried? If so why does profile Item have a general error message?

Reply:
I reaslied this as well. For commands that are without the letter specifier, like 
`view me 1`
`delete me 1`
It seems to have this special other error message generator.

Reply:
Is there anywhere we can consolidate this issue. This is linked to the problem Keane and I was discussing in the post above. @AY2021S1-CS2103T-T15-4/developers

Reply:
In reference to my earlier post, what is the point of the switch statement if all of them are fallthroughs?

Reply:
Since many of our commands require the `parseIndex` to throw a specific message, we could pass in the specific error message in the `parseIndex` function. And to fix this issue you pointed out, perhaps we could create individual parsers for each of the commands, so that they can throw their own specific error message.

Reply:
So I take that we should format it such that delete, view, find should follow how add, edit does it?

Reply:
Updates to be made

**For view and delete**
@orzymandias @shawn-nyk @keanecjy
* Abstract out the viewcommandParser into the data types just like how we did it in add and edit.
* Abstract out the deleteCommandParser into the data types just like how we did it in add and edit.

**Notes**
* For `delete` and `view` I noticed that we have a method called `isValidType`, I wonder if there is even a need for them. For the switch case, doesn't it already does the job of checking whether the type is valid?
* We can do the `checkCommandDetailsIsNotBlank` in the individual parser itself instead.

Reply:
Note that the current push is invalid, it will cause the wrong message due to the structure